### DISCUSSION TITLE
Why this Svg Path element does not works as I expect it to ?

### DISCUSSION THREAD

**Discussion by loloof64** [slint-ui/slint#4205] [Q&A] [answered]
Hi ! I'm experimenting with the Path widget in slint editor : [My experiment](https://slint.dev/releases/1.3.2/editor/?snippet=export+component+Example+inherits+Rectangle+%7B%0A++++width%3A+100px%3B%0A++++height%3A+100px%3B%0A++++border-color%3A+blue%3B%0A++++border-width%3A+1px%3B%0A++++Path+%7B%0A++++++++width%3A+100px%3B%0A++++++++height%3A+100px%3B%0A++++++++commands%3A+%22M+0+0+L+0+100%22%3B%0A++++++++stroke%3A+red%3B%0A++++++++stroke-width%3A+1px%3B%0A++++%7D%0A%7D%0A&style=native).

Here's the code

```javascript
export component Example inherits Rectangle {
    width: 100px;
    height: 100px;
    Path {
        width: 100px;
        height: 100px;
        commands: "M 0 0 L 0 100";
        stroke: red;
        stroke-width: 1px;
    }
}
```

What I'm expecting is that the red line inside the rectangle is "appended" to the left side of the rectangle, as I'm defining a vectical line with both points abciss defined to 0.

But instead, the line is centered horizontally in the rectangle : why ?

Also the same for [horizontal line](https://slint.dev/releases/1.3.2/editor/?snippet=export+component+Example+inherits+Rectangle+%7B%0A++++width%3A+100px%3B%0A++++height%3A+100px%3B%0A++++border-color%3A+blue%3B%0A++++border-width%3A+1px%3B%0A++++Path+%7B%0A++++++++width%3A+100px%3B%0A++++++++height%3A+100px%3B%0A++++++++commands%3A+%22M+0+0+L+100+0%22%3B%0A++++++++stroke%3A+red%3B%0A++++++++stroke-width%3A+1px%3B%0A++++%7D%0A%7D%0A&style=native).

This is the code this time
```javascript
export component Example inherits Rectangle {
    width: 100px;
    height: 100px;
    border-color: blue;
    border-width: 1px;
    Path {
        width: 100px;
        height: 100px;
        commands: "M 0 0 L 100 0";
        stroke: red;
        stroke-width: 1px;
    }
}
```

**Comment by loloof64** [comment]
After playing more with my playground, it seems that whatever I "draw", it will be centered in the rectangle. So, as I also need to draw a more complex path, I'm trying to adapt the position and the commands coordinates in order to achive my goal.

In fact, I'm just trying to draw a simple arrow between two points (defined dynamically) in my custom component.

**Comment by Vadoola** [comment, marked as answer]
Try setting the viewbox height and width to the same as the widget height and width: as [seen here](https://slint.dev/releases/1.3.2/editor/?snippet=export+component+Example+inherits+Rectangle+%7B%0A++++width%3A+100px%3B%0A++++height%3A+100px%3B%0A++++border-color%3A+blue%3B%0A++++border-width%3A+1px%3B%0A++++Path+%7B%0A++++++++width%3A+100px%3B%0A++++++++height%3A+100px%3B%0A++++++++viewbox-height%3A+100%3B%0A++++++++viewbox-width%3A+100%3B%0A++++++++commands%3A+%22M+0+0+L+0+100%22%3B%0A++++++++stroke%3A+red%3B%0A++++++++stroke-width%3A+5px%3B%0A++++%7D%0A%7D%0A&style=native).

I'm not an expert on paths, and SVGs (which I know the Slint paths are based on) to know the specific details of why this is the way it is, but I know this works from when I was running into the same issue, while I was first learning Slint.

**Reply by loloof64** [reply]
Thank you very much 😃
